client/my-sites/stats/controller.js

```diff
@@ -188,7 +189,12 @@ module.exports = {
 
 		analytics.pageView.record( basePath, analyticsPageTitle + ' > Insights' );
 
-		if ( site.post_count <= 1 &&
+		// Calculate the number of posts created by the user (not Headstarted) to determine if we should show the Insights tab, or the NUX placeholder.
+		postCount = site.post_count;
+		if ( site.options.headstart ) {
```

Comment:
I don't really know how the `site` object is constructed, but to be on the safe side (like they do on line 171 above) you might want to also test for `site.options`, like `if ( site.options && site.options.headstart )`.